### PR TITLE
Copy adding functionalities

### PR DESCRIPTION
**Pull request by amalexandrescu**
Copy of the last branch which was stable and adding more frontend functionalities

### CODE REVIEW

File: src/components/TableCreation.tsx [hunk 1]
```diff
@@ -123,7 +123,8 @@ const TableCreation = ({ rows, rowContent, columns, tableId, customTableWidth }:
     return sortedRows;
   }, [rows, sortedColumn, sortDirection]);
 
-  const rowHeight = tableId === 'waiting' || tableId === 'miners' ? 64.8 : 52.813; // row height per table ids, to get table's height
+  const rowHeight =
```
Comment (BowTiedDeployer) [inline src/components/TableCreation.tsx:126]: this id is needed to take the corresponding row height based on what type of table it is.

File: src/components/appMenuSections/dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -21,8 +20,8 @@ import { Box } from '@mui/material';
 const Dashboard = () => {
   const [authenticatedSuccessfully, setAuthenticatedSuccessfully] = useState<boolean>(false);
   const [clickedJoinPoolButtonByViewer, setClickedJoinPoolButtonByViewer] = useState<boolean>(false);
-  const [currentNotifier, setCurrentNotifier] = useState<string>('');
-  const [minersList, setMinersList] = useState<any>([]);
+  const [currentNotifier, setCurrentNotifier] = useState<string | null>(null);
```
Comment (BowTiedDeployer) [inline src/components/appMenuSections/dashboard/Dashboard.tsx:23]: is it the best way to initialize it? (before taking the value from the smart contract request)

File: src/components/appMenuSections/profile/MinerProfile.tsx [hunk 3]
```diff
@@ -195,7 +199,7 @@ const MinerProfile = () => {
             type="number"
             onChange={(e) => {
               const inputAmount = e.target.value;
-              const inputAmountToInt = parseInt(inputAmount);
+              const inputAmountToInt = parseFloat(inputAmount);
```
Comment (BowTiedDeployer) [inline src/components/appMenuSections/profile/MinerProfile.tsx:202]: why is it named "ToInt". it refers to the amount of stx.

rename it. by doing it here, would appear conflicts in the following merges, should we only rename in the last commit? @Alexandrescu

File: src/components/appMenuSections/profile/MinerProfile.tsx [hunk 4]
```diff
@@ -216,7 +219,7 @@ const MinerProfile = () => {
...
```
Comment (BowTiedDeployer) [inline src/components/appMenuSections/profile/MinerProfile.tsx:222]: same here

> 
> why is it named "ToInt". it refers to the amount of stx.
> 
> rename it. by doing it here, would appear conflicts in the following merges, should we only rename in the last commit?

**Review by BowTiedDeployer — DISMISSED** [review]

File: src/components/appMenuSections/dashboard/Dashboard.tsx (hunk 2 above)
Comment (Alexandrescu) [reply src/components/appMenuSections/dashboard/Dashboard.tsx:23]: We'd need to have a way of doing an async loading for some of these. https://github.com/stacks-degens/starters-front-end/issues/55

File: src/components/appMenuSections/profile/MinerProfile.tsx (hunk 3 above)
Comment (Alexandrescu) [reply src/components/appMenuSections/profile/MinerProfile.tsx:202]: https://github.com/stacks-degens/starters-front-end/issues/56

File: src/components/appMenuSections/profile/WaitingMinerProfile.tsx [hunk 5]
```diff
@@ -20,11 +20,12 @@ const WaitingMinerProfile = () => {
...
   useEffect(() => {
     const fetchData = async () => {
-      const newWaitingList = await ReadOnlyAllDataWaitingMiners(userAddressAsCV);
-      setPositiveVotes(newWaitingList[0].value[0].value.value['positive-votes'].value);
-      setPositiveVotesThreshold(newWaitingList[0].value[0].value.value['positive-threshold'].value);
-      setNegativeVotes(newWaitingList[0].value[0].value.value['negative-votes'].value);
-      setNegativeVotesThreshold(newWaitingList[0].value[0].value.value['negative-threshold'].value);
+      const waitingList = await ReadOnlyAllDataWaitingMiners(userAddressAsCV);
+      const newWaitingList = cvToJSON(waitingList.newResultList[0]);
+      setPositiveVotes(newWaitingList.value[0].value.value['pos-votes'].value);
```
Comment (Alexandrescu) [inline src/components/appMenuSections/profile/WaitingMinerProfile.tsx:25]: You could've had only one variable here instead of moving 4 items to the state.

Comment (Alexandrescu) [reply src/components/appMenuSections/profile/WaitingMinerProfile.tsx:25]: https://github.com/stacks-degens/starters-front-end/issues/57

**Review by Alexandrescu — APPROVED** [review]

**Pull request merged**